tests/e2e/kserve_test.go

```diff
@@ -511,3 +515,146 @@ func (tc *KserveTestCtx) createConnectionSecret(secretName, namespace string) {
+func (tc *KserveTestCtx) ValidateDSCIDSCValidationInteractionForKserve(t *testing.T) {
+	t.Helper()
+
+	jqKserveReadyFalse := `
+	.status.conditions[]
+	| select(.type == "KserveReady" and .status == "%s")
+	| .message
+	| test("servicemesh needs to be set to 'managed' in dsci cr"; "i")`
+
+	t.Log("Disabling ServiceMesh in DSCI")
+	tc.EnsureResourceCreatedOrPatched(
+		WithMinimalObject(gvk.DSCInitialization, tc.DSCInitializationNamespacedName),
+		WithMutateFunc(testf.Transform(
+			`.spec.serviceMesh.managementState = "%s"`, operatorv1.Removed,
+		)),
+	)
+
+	t.Log("Verifying KServe reports dependency on ServiceMesh correctly")
+	tc.EnsureResourceExists(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithCondition(jq.Match(jqKserveReadyFalse, metav1.ConditionFalse)),
+	)
+
+	t.Log("Re-enabling ServiceMesh in DSCI for recovery")
+	tc.EnsureResourceCreatedOrPatched(
+		WithMinimalObject(gvk.DSCInitialization, tc.DSCInitializationNamespacedName),
+		WithMutateFunc(testf.Transform(
+			`.spec.serviceMesh.managementState = "%s"`,
+			operatorv1.Managed,
+		)),
+	)
+
+	t.Log("Verifying KServe becomes ready after ServiceMesh is enabled")
+	tc.verifyKserveStatus(t, metav1.ConditionTrue)
+
+	t.Log("DSCI/DSC validation interaction test completed successfully")
+}
+
+// ValidateCustomCertificateCreation tests that a valid custom certificate is created for OpenshiftDefaultIngress.
+func (tc *KserveTestCtx) ValidateCustomCertificateCreation(t *testing.T) {
+	t.Helper()
+
+	originalDSCI := tc.FetchDSCInitialization()
+	customSecretName := "custom-test-secret"
+
+	t.Log("Configuring Kserve with OpenshiftDefaultIngress and custom secret")
+	tc.EnsureResourceCreatedOrPatched(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithMutateFunc(testf.TransformPipeline(
+			testf.Transform(`.spec.components.kserve.serving.ingressGateway.certificate.secretName = "%s"`, customSecretName),
+		)),
+	)
+
+	t.Log("Verifying Kserve is ready")
+	tc.verifyKserveStatus(t, metav1.ConditionTrue)
+
+	t.Log("Verifying custom secret is created")
+	tc.EnsureResourceExists(
+		WithMinimalObject(gvk.Secret, types.NamespacedName{Namespace: originalDSCI.Spec.ServiceMesh.ControlPlane.Namespace, Name: customSecretName}),
+	)
+
+	t.Log("Deleting secretName from DSC and verifying Kserve readiness")
+	tc.EnsureResourceCreatedOrPatched(
+		WithMinimalObject(gvk.DataScienceCluster, tc.DataScienceClusterNamespacedName),
+		WithMutateFunc(testf.Transform(`.spec.components.kserve.serving.ingressGateway.certificate |= del(.secretName)`)),
+	)
+	tc.verifyKserveStatus(t, metav1.ConditionTrue)
+
+	t.Log("Deleting custom secret")
+	tc.DeleteResource(
+		WithMinimalObject(
+			gvk.Secret,
+			types.NamespacedName{Namespace: originalDSCI.Spec.ServiceMesh.ControlPlane.Namespace, Name: customSecretName},
+		),
+		WithWaitForDeletion(true),
+	)
+
+	t.Log("Custom certificate creation test completed successfully")
```

Comment:
```suggestion
```
These completion logs at the end of each test function can be removed. They don't add debugging value, and the test framework already indicates success/failure.